Automattic/wp-calypso
# Use new woo eligibility hook in woo install landing page

#### Changes proposed in this Pull Request

On this PR:

* Remove the site plan handling since it's going to be handled straight in the woop installer flow (https://github.com/Automattic/wp-calypso/pull/58602)
* Checks whether the site has transferring blockers. If so, it disables the setup button.
* Point the setup button to the new flow when `woop` config is enabled


https://user-images.githubusercontent.com/77539/144259700-d32e0c92-7812-4808-9d47-532a7efb350b.mov


#### Testing instructions



* Confirm the landing page won't show the upgrade variation, even when the site doesn't have a proper plan

before | after
-----|------
![image](https://user-images.githubusercontent.com/77539/144239137-ed4645ee-f441-40d2-932c-8a3a95a790bb.png) | ![image](https://user-images.githubusercontent.com/77539/144239192-67df3a7b-f007-4be0-9643-20d1a1210478.png)


* Confirm the setup button is disabled when the site has some transferring issues. For this:
  * Sandbox the REST API
  * Apply this patch D70717-code (generates some eligibility warnings and errors)
<img width="500" alt="Screen Shot 2021-12-01 at 9 59 51 AM" src="https://user-images.githubusercontent.com/77539/144238867-ea73c138-5461-4263-ad31-c41be5e65a71.png">

* Navigate in between all steps, from the landing page to the transferring step. Check the navigation works as expected.



Related to #
